pytorch_lightning/trainer/trainer.py

```diff
@@ -791,44 +844,60 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
             model.load_state_dict(ckpt['state_dict'])
 
         # attach dataloaders
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
+            kwargs = {'test_dataloaders' if stage == 'test' else 'val_dataloaders': dataloaders}
+            self.data_connector.attach_dataloaders(model, **kwargs)
 
         # run tests
-        self.tested_ckpt_path = ckpt_path
-        self.testing = True
-        os.environ['PL_TESTING_MODE'] = '1'
+        self.evaluating = stage
+        self.evaluated_ckpt_path = ckpt_path
         self.model = model
         results = self.fit(model)
-        self.testing = False
-        del os.environ['PL_TESTING_MODE']
+        self.evaluating = None
 
         # teardown
         if self.is_function_implemented('teardown'):
             model_ref = self.get_model()
-            model_ref.teardown('test')
+            model_ref.teardown(stage)
 
         return results
 
-    def __test_given_model(self, model, test_dataloaders):
+    def __evaluate_given_model(self, model, dataloaders, stage: str):
 
         # attach data
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
+            kwargs = {'test_dataloaders' if stage == 'test' else 'val_dataloaders': dataloaders}
+            self.data_connector.attach_dataloaders(model, **kwargs)
 
         # run test
         # sets up testing so we short circuit to eval
-        self.testing = True
+        self.evaluating = stage
         self.model = model
         results = self.fit(model)
-        self.testing = False
+        self.evaluating = None
 
         # teardown
         if self.is_function_implemented('teardown'):
-            model.teardown('test')
+            model.teardown(stage)
 
         return results
 
+    @property
+    def testing(self):
+        warnings.warn(
+            'Trainer.testing is deprecated, use Trainer.evaluating instead.',
+            FutureWarning, stacklevel=2
+        )
+        return bool(self.evaluating)
+
+    @property
+    def tested_ckpt_path(self):
+        warnings.warn(
+            'Trainer.tested_ckpt_path is deprecated and has been replaced by Trainer.evaluated_ckpt_path.',
+            FutureWarning, stacklevel=2
+        )
+        return self.evaluated_ckpt_path
```

Comment:
Ah, I'd like your input on these deprecation messages:
* Do you think they should be `DeprecationWarning`s or `FutureWarning`s, so visible by default on the terminal?
* I saw that in other deprecation messages you mention the version in which it's supposed to be removed, should I add something here?

Reply:
`DeprecationWarning` of course. Deprecate in v1.1 and remove in v1.3

Reply:
Thanks